xdsl/dialects/x86.py

```diff
@@ -0,0 +1,141 @@
+from __future__ import annotations
+
+from abc import ABC, abstractmethod
+from typing import ClassVar
+
+from typing_extensions import Self
+
+from xdsl.ir import (
+    Data,
+    Dialect,
+    TypeAttribute,
+)
+from xdsl.irdl import (
+    irdl_attr_definition,
+)
+from xdsl.parser import AttrParser
+from xdsl.printer import Printer
+from xdsl.utils.exceptions import VerifyException
+
+
+class X86RegisterType(Data[str], TypeAttribute, ABC):
```

Comment:
I'm wondering whether it would make sense to move the shared functionality to a `RegisterType` abstract base class in the `backend` directory, used by all the assembly dialects. I have a feeling that this would make a lot of things simpler, but I don't know how it impacts the other code in your branch. Could you please have a go at this?

Reply:
Just so there's no confusion, could you please do this in this PR?

Reply:
Yes, maybe as part of this PR, or as a separate PR to be merged before this one. TBH I think it's a small enough change that it can be done in this PR.